systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java

```diff
@@ -108,12 +112,56 @@ protected static JsonArray readUpgradeJson(String fileName) {
             procedures.put("interBrokerProtocolVersion", testKafkaVersion.protocolVersion());
             data.put("proceduresAfterOperatorUpgrade", procedures);
 
-            parameters.add(Arguments.of(data.getString("fromVersion"), "HEAD", data));
+            // fromVersion is the mid step here, for CO upgrade we'll use this: 'prevVersion' -> 'fromVersion' -> HEAD
+            parameters.add(Arguments.of(data.getString("prevVersion"), data.getString("fromVersion"), "HEAD", data));
         });
 
         return parameters.stream();
     }
 
+    protected static List<JsonObject> buildMidStepUpgradeData(JsonObject jsonData) {
+        List<TestKafkaVersion> testKafkaVersions = TestKafkaVersion.getKafkaVersions();
+        TestKafkaVersion testKafkaVersion = testKafkaVersions.get(testKafkaVersions.size() - 1);
+
+        List<JsonObject> steps = new ArrayList<>();
```

Comment:
Maybe would be better to have Map instead of array? To avoid magic numbers later in getting data?